---
 fix custom sessions (see bug #216984, work done by Gilles, I'm just committing)

 config/Xsession.in |   18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

Index: b/config/Xsession.in
===================================================================
--- a/config/Xsession.in
+++ b/config/Xsession.in
@@ -177,15 +177,6 @@ if [ -n "$GDM_LANG" ]; then
   fi
 fi
 
-# run all system xinitrc shell scripts.
-if [ -d /etc/X11/xinit/xinitrc.d ]; then
-    for i in /etc/X11/xinit/xinitrc.d/* ; do
-        if [ -x "$i" ]; then
-	    . "$i"
-        fi
-    done
-fi
-
 if [ "x$command" = "xcustom" ] ; then
   if [ -x "$HOME/.xsession" ]; then
     command="$HOME/.xsession"
@@ -213,6 +204,15 @@ if [ "x$command" = "xdefault" ] ; then
   fi
 fi
 
+# run all system xinitrc shell scripts.
+if [ -d /etc/X11/xinit/xinitrc.d ]; then
+    for i in /etc/X11/xinit/xinitrc.d/* ; do
+        if [ -x "$i" ]; then
+	    . "$i"
+        fi
+    done
+fi
+
 # add ssh-agent if found
 sshagent="`gdmwhich ssh-agent`"
 if [ -n "$sshagent" ] && [ -x "$sshagent" ] && [ -z "$SSH_AUTH_SOCK" ]; then
